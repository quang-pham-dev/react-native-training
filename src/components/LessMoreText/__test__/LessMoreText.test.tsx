import React from 'react'

// LIBS
import {render} from '@testing-library/react-native'

// Components
import MoreLessText from '@components/LessMoreText'

describe('Read more Component', () => {
  const props = {
    numberOfLines: 2,
    styleShowMoreText: {color: 'red'},
  }

  test('should render correctly', () => {
    const tree = render(<MoreLessText {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test('should render correctly with children', () => {
    const tree = render(<MoreLessText {...props}>Hello</MoreLessText>).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
