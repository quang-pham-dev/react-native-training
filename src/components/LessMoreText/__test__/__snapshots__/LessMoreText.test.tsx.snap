// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Read more Component should render correctly 1`] = `
<Text
  onTextLayout={[Function]}
  style={
    Array [
      Object {
        "color": "#8F959E",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 21,
      },
    ]
  }
/>
`;

exports[`Read more Component should render correctly with children 1`] = `
<Text
  onTextLayout={[Function]}
  style={
    Array [
      Object {
        "color": "#8F959E",
        "fontFamily": "Inter-Regular",
        "fontSize": 15,
        "lineHeight": 21,
      },
    ]
  }
>
  Hello
</Text>
`;
